common/util.go

```diff
@@ -97,7 +97,7 @@ const (
 	taskNotReadyRescheduleBackoffCoefficient = 1.5
 	taskNotReadyRescheduleMaxInterval        = 3 * time.Minute
 
-	dependencyTaskNotCompletedRescheduleInitialInterval    = 10 * time.Second
+	dependencyTaskNotCompletedRescheduleInitialInterval    = 3 * time.Second
```

Comment:
We should probably make this interval depend on the AckLevelUpdateInterval dynamic config. and for integration test, override the dynamic config to use a smaller value (this is already the case I think).

I don't have very strong opinion for now as in most cases the close task should be long gone when the deletion task is executed, so couple more retry attempts should be ok. Maybe leave a comment here so that we can re-visit this if it becomes a problem when deleting namespace/workflow.

Reply:
I tried making this dependent on the dynamic config, but it ended up being a huge lift. Instead, I've added a comment.